File: JAVA/3254. Find the Power of K-Size Subarrays I.java
class Solution {
    public int[] resultsArray(int[] n, int k) {
        int[] a=new int[n.length-k+1];
        for(int i=0;i<=n.length-k;i++){
            boolean t=true;
            int m=n[i];
            for(int j=1;j<k;j++){
                if(n[i+j]!=n[i+j-1]+1){
                    t=false;
                    break;
                }
                m=n[i+j];
                }
                if(t==true)
                a[i]=m;
                else
                a[i]=-1;
            
        }
        return a;
    }
}

File: C++/3254. Find the Power of K-Size Subarrays I.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& n, int k) {
        vector<int> r;
        for (int i = 0; i <= n.size() - k; i++) {
            bool t = true;
            int m = n[i]; 
            for (int j = 1; j < k; j++) {
                if (n[i + j] != n[i + j - 1] + 1) {
                    t = false;
                    break;
                }
                m = n[i + j]);
            }
            if (t) {
                r.push_back(m);
            } else {
                r.push_back(-1);
            }
        }
        
        return r;
    }
};
